src/components/Profile/NFTCollectionList.tsx

```diff
@@ -0,0 +1,106 @@
+'use client';
+
+import { createIndicator } from '@masknet/shared-base';
+import { useSuspenseInfiniteQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+
+import { GridListInPage } from '@/components/GridListInPage.js';
+import { Image } from '@/components/Image.js';
+import { POAPGridListComponent } from '@/components/Profile/POAPList.js';
+import { classNames } from '@/helpers/classNames.js';
+import { FireflySocialMediaProvider } from '@/providers/firefly/SocialMedia.js';
+import type { Collection } from '@/providers/types/Firefly.js';
+
+export interface NFTCollectionItemProps {
+    collection: Collection;
+    onClick?: (collection: Collection) => void;
+}
+
+export function NFTCollectionItem({ collection, onClick }: NFTCollectionItemProps) {
+    const distinctNFTCount = collection.collection_details.distinct_nft_count;
+    const images = useMemo(() => {
+        if (!collection.nftPreviews) return [];
+        let previewImages = collection.nftPreviews;
+        if (collection.nftPreviews.length === 4 && distinctNFTCount === 4) {
+            previewImages = previewImages?.slice(0, 4);
+        } else {
+            previewImages = previewImages?.slice(0, 3);
```

Comment:
```suggestion
            previewImages = previewImages.slice(0, 4);
        } else {
            previewImages = previewImages.slice(0, 3);
```
It has already been checked above.